src/app/broadcast.js

```diff
@@ -82,7 +82,7 @@ class Broadcast {
      * Checks if the signal is pristine, so no interferences happened
      */
     isGood() {
-        return this.state === states.BROADCASTING || this.state === states.FINISHED;
+        return (this.state === states.BROADCASTING || this.state === states.FINISHED) && this.interactedBroadcasts.length === 0;
```

Comment:
This makes sense, but the root cause problem is that `this.interactedBroadcasts.length` != 0 and `this.state` is wrong. Wouldn't it be better to solve that root cause instead?